spinnaker_camera_driver/src/nodelet.cpp

```diff
@@ -770,6 +779,30 @@ class SpinnakerCameraNodelet : public nodelet::Nodelet {
     return true;
   }
 
+  void publishMonoImage(const sensor_msgs::ImagePtr& image,
+                        const sensor_msgs::CameraInfoPtr& ci) {
+    if (!publish_mono_) {
```

Comment:
I would maybe move this if-clause outside of the function so that function is only called when publish_mono_ is true.